Tools/CISupport/ews-app/ews/common/github.py

```diff
@@ -215,7 +215,8 @@ def generate_updated_pr_description(self, description, ews_comment):
         return u'{}{}\n{}\n{}'.format(description, self.STATUS_BUBBLE_START, ews_comment, self.STATUS_BUBBLE_END)
 
     def generate_comment_text_for_change(self, change):
-        hash_url = 'https://github.com/{}/commit/{}'.format(change.pr_project, change.change_id)
+        repository_url = 'https://github.com/{}'.format(change.pr_project)
+        hash_url = '{}/commit/{}'.format(repository_url, change.change_id)
```

Comment:
Let's hope those string values never need to be URL-encoded.  :)

Reply:
They shouldn't...we're dealing with them as these intermediate strings because it's easier, but we could imagine a world where we constructed them at the very last moment. `change_id` is always guaranteed to be a hash, and `pr_project` is ultimately a string we control (also, there are restrictions from GitHub on that string, although I'm not familiar with what they care, they presumably forbid certain characters, like `/`)